R/resource.R

```diff
@@ -0,0 +1,160 @@
+##' Specify what resources a task requires to run.
+##'
+##' # Windows cluster
+##'
+##' The windows cluster currently is all 32-core nodes, with RAM
+##' mainly 512Gb, and a few 384Gb.
+##'
+##'
+##' # Linux cluster
+##'
+##' Coming Soon.
+##'
+##' @title Task Resources
+##'
+##' @param cores The number of cores your task requires. This is 1 by
+##'   default. An error is reported if you request more cores than
+##'   is available on the largest cluster node. Setting to `Inf` will
+##'   request any entire node, however many cores it has.
+##'
+##' @param exclusive Set this to `TRUE` to ensure no other tasks will be
```

Comment:
Add something about the intent? I know we had a couple of different use cases here.

> This might be useful in the case where your task will use so much RAM that we don't want anything else on the same node, or where ...